src/foam/java/refinements.js

```diff
@@ -435,10 +435,10 @@ foam.CLASS({
       var m = info.getMethod('cast');
       m.body = 'return ( o instanceof Number ) ?'
             + '((Number)o).intValue() :'
-            + '(int)o;';
+            + '((Integer) o).intValue();';
```

Comment:
I'm not sure the old code was ever being triggered, it doesn't look like it should, if your parameter is Object java will automatically box primitive types for you.

I think we should just remove the conditional and do `((Number)o).intValue()` and similar for all cases.  No sense updating and keeping a conditional that literally never happens.